scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,687 @@
+        printMsgs "console" "INFO: Moving the Local DB and Import folders to new configuration folder (total: $f_size Mb)"
+
+        local folder
+        for folder in dbs import; do
+            mv "$home/.skyscraper/$folder" "$home/.skyscraper-$folder" && \
+                printMsgs "console" "INFO: Moved "$home/.skyscraper/$folder" to "$home/.skyscraper-$folder""
+        done
+
+        # When having an existing installation, changes are the gamelist is in the ROMs folder
+        # Create a GUI config file with this setting pre-set.
+        iniConfig " = " '"' "$configdir/all/skyscraper.cfg"
+        iniSet "use_rom_folder" 1
+    fi
+
+    moveConfigDir "$home/.skyscraper" "$configdir/all/skyscraper"
+
+    # When migrating an existing installation, make sure we move the Local DB and Import folder back the new conf folder
+    for folder in dbs import; do
+        if [[ -d "$home/.skyscraper-$folder" ]]; then
+            printMsgs "console" "INFO: Moving "$home/.skyscraper-$folder" back to configuration folder"
+            mv  "$home/.skyscraper-$folder" "$configdir/all/skyscraper/$folder"
+        fi
+    done
+
+    _init_config_skyscraper
+    chown -R $user:$user "$configdir/all/skyscraper"
+}
+
+function _init_config_skyscraper() {
+    local md_conf_dir="$configdir/all/skyscraper"
+
+    cat > "$md_conf_dir/config.ini.rp-dist" << _EOF_
+[main]
+frontend="emulationstation"
+dbFolder="$md_conf_dir/dbs"
+importFolder="$md_conf_dir/import"
+inputFolder="$romdir"
+unattend="true"
+unattendSkip="true"
+noHints="false"
+threads="2"
+maxFails="30"
+lang="en"
+region="wor"
+videos="false"
+marquees="true"
+wheels="false"
+covers="true"
+screenshots="true"
+artworkXml="$md_conf_dir/artwork.xml"
+
+_EOF_
+
+    # Make sure the `artwork.xml` and other conf file(s) are present, but don't overwrite them on upgrades.
+    # For now we just have one file, but more might come up in the future.
+    local f_conf
+    for f_conf in artwork.xml; do
+        if [[ -f "$md_conf_dir/$f_conf" ]]; then
+            cp "$md_build/$f_conf" "$md_conf_dir/$f_conf.rp-dist"
+        else
+            cp "$md_build/$f_conf" "$md_conf_dir"
+        fi
+    done
+
+    # If we don't have a previous config.ini file, copy the dist one
+    [[ ! -f "$md_conf_dir/config.ini" ]] && cp "$md_conf_dir/config.ini.rp-dist" "$md_conf_dir/config.ini"
+
+    # Try to find the rest of the necesary files from the qmake build file
+    # They should be listed in the `unix:examples.file` configuration line
+    if [[ $(grep unix:examples.files "$md_build/skyscraper.pro" 2>/dev/null | cut -d= -f2-) ]]; then
+        local files=$(grep unix:examples.files "$md_build/skyscraper.pro" | cut -d= -f2-)
+        local file
+
+        for file in $files; do
+            # Copy the files to the configuration folder. Skip config.ini and artwork.xml, they're already handled
+            if [[ $file != "artwork.xml"  && $file != "config.ini" ]]; then
```

Comment:
very minor but I noticed you have 2 spaces before &&

Reply:
OK, will fix it.